### PR TITLE
Also adds SqlStatementParameterCustomizer

### PR DESCRIPTION
**Pull request by stevenschlansker**
I was still porting forward the work I did on #648 
I am not sure how to reconcile with #711 

Bit rough as it was a WIP.  I can clean up if anything will end up being committed.

### CODE REVIEW

**Comment by qualidafial** [comment]
I lifted SqlStatementParameterCustomer from your previous branch so my changes should be compatible with your branch.

Sorry for the duplicate work.

File: sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/CustomizingStatementHandler.java [hunk 1]
```diff
@@ -13,97 +13,101 @@
...
-                // There are no customizing annotations on the parameter, so use default binder
-                paramBasedCustomizerFactories.add(new FactoryAnnotationParameterIndex(new Bind.Factory(), null, parameters[paramIndex], paramIndex));
+        final Function<Integer, Stream<BoundCustomizer>> customizeForParameter =
+                i -> defaultBinding(annotationsFor(parameters[i]))
+                        .map(a -> instantiateFactory(a).createForParameter(a, type, method, parameters[i], i))
+                        .map(c -> (stmt, args) -> c.apply(stmt, args[i]));
+
+        return IntStream.range(0, parameters.length)
+                .mapToObj(Integer::valueOf)
+                .flatMap(customizeForParameter);
```
Comment (qualidafial) [inline sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/CustomizingStatementHandler.java]: Really nice cleanup here

**Comment by qualidafial** [comment]
@stevenschlansker If you want, I can comb through the changes in my PR and submit a new one against yours that replicates any changes I'd like to keep.

**Comment by stevenschlansker** [comment]
Sure, I'll work on getting this merged tonight, then maybe you can file it as a followup?

**Pull request merged**